File: src/Samples/Scheduler.ASP.NET.VB/Data.ashx.vb
Imports System.Web
Imports System.Web.Services
Imports DHTMLX.Scheduler
Imports DHTMLX.Scheduler.Controls
Imports DHTMLX.Scheduler.Data
Imports DHTMLX.Common


Public Class Data
    Implements System.Web.IHttpHandler

    Sub ProcessRequest(ByVal context As HttpContext) Implements IHttpHandler.ProcessRequest


        Dim dataResult As SchedulerAjaxData
        If context.Request.QueryString("recurring") IsNot Nothing Then
            dataResult = New SchedulerAjaxData((New SchedulerDataContext()).Recurrings)
        Else
            dataResult = New SchedulerAjaxData((New SchedulerDataContext()).Events)
        End If
        context.Response.ContentType = "application/json"
        context.Response.Write(dataResult.ToString())

    End Sub

    ReadOnly Property IsReusable() As Boolean Implements IHttpHandler.IsReusable
        Get
            Return False
        End Get
    End Property

End Class

File: src/Samples/Scheduler.ASP.NET/Data.ashx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

using DHTMLX.Scheduler.Data;

namespace SchedulerNetAsp
{
   
    /// <summary>
    /// </summary>
    public class Data : IHttpHandler
    {

        public void ProcessRequest(HttpContext context)
        {
            SchedulerAjaxData data;
            var dc = new SchedulerDataContext();

            if (context.Request.QueryString["recurring"] == null)
                data = new SchedulerAjaxData(dc.Events);
            else
                data = new SchedulerAjaxData(dc.Recurrings);

            context.Response.ContentType = "text/json";
            context.Response.Write(data.ToString());
        }

        public bool IsReusable
        {
            get
            {
                return false;
            }
        }
    }
}